pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -215,7 +215,8 @@
             prepareDynamicConfigurationMap();
     private final ConcurrentOpenHashMap<String, Consumer<?>> configRegisteredListeners;
 
-    private final ConcurrentLinkedQueue<Pair<String, CompletableFuture<Optional<Topic>>>> pendingTopicLoadingQueue;
+    private final ConcurrentLinkedQueue<Triple<String, CompletableFuture<Optional<Topic>>, Map<String, String>>>
```

Comment:
Maybe we could add the properties map in class `Topic`?

Reply:
Only PersistentTopic has the properties.

Reply:
I think using `Triple` harms the code readability. I'd rather like adding a simple class

```java
    @AllArgsConstructor
    private static class TopicLoadingContext {
        private final String name;
        private final CompletableFuture<Optional<Topic>> topicFuture;
        private final Map<String, String> properties;
    }
```

We don't need the `Triple` here because we don't use it as a key of `Map` so we don't need the `compareTo`, `equals` and `hashCode` methods that `Triple` has implemented by default.

Reply:
Ok, good suggestion.